src/components/ide.js

```diff
@@ -1,40 +1,91 @@
-import React, { useEffect } from 'react';
-import { UnControlled as CodeMirror } from "react-codemirror2";
-import { useContext } from "react";
-import { HeartwoodStateContext, HeartwoodDispatchContext } from "../state/HeartwoodContextProvider";
-import '../styles/ide.css';
+import React, { useEffect, useRef, useContext, useState } from "react"
+import firebase from "gatsby-plugin-firebase"
+import { HeartwoodStateContext, HeartwoodDispatchContext } from "../state/HeartwoodContextProvider"
+import "../styles/ide.css"
 
+import { getEndpointPrefix } from "../utils/utils"
 
 const IDE = () => {
- // const [socketUrl, setSocketUrl] = useState("ws://localhost:8080");
- // const [sendMessage, lastMessage, readyState, getWebSocket] = useWebSocket(
- //   socketUrl
-  //);
-  //const handleClickSendMessage = useCallback((val) => sendMessage(val), []);
-  const state = useContext(HeartwoodStateContext);
-  const dispatch = useContext(HeartwoodDispatchContext);
-  
-  useEffect( () => {
-    require('codemirror/mode/python/python'); // need to rerender CodeMirror after this to enable syntax highlighting
-  }, [])
+  const editorRef = useRef(null)
+
+  const state = useContext(HeartwoodStateContext)
+  const dispatch = useContext(HeartwoodDispatchContext)
+  const [ref, setRef] = useState("")
+  const [input, setInput] = useState("blank")
```

Comment:
Codacy found an issue: ['input' is assigned a value but never used.](https://app.codacy.com/gh/treetoplearning/heartwood/pullRequest?prid=5919976)